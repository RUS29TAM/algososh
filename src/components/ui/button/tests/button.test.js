import renderer from 'react-test-renderer'
import {render, screen, fireEvent} from "@testing-library/react";
import {Button} from "../button";

describe('Тест для компонента Button', () => {
    it('Компонент Button отрисовка текста компонента - тест пройден', () => {
        const button = renderer.create(<Button text={'text'}/>).toJSON();
        expect(button).toMatchSnapshot();
    })
})

describe('Тест для слушателей событий компонента Button', () => {
    it('Нажатие на кнопку работает - тест пройден', () => {
        const onClick = jest.fn();
        render(<Button text={'text'} onClick={onClick}/>)
        const button = screen.getByText('text');
        fireEvent.click(button)
    })
})