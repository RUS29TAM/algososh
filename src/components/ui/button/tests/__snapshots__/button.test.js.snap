// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест для компонента Button Компонент Button отрисовка текста компонента - тест пройден 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;
